### ISSUE TITLE
crash on update

### ISSUE THREAD

**Issue by Kamik423** [MacDownApp/macdown#900]
<img width="532" alt="screen shot 2017-12-12 at 3 38 29 pm" src="https://user-images.githubusercontent.com/16278233/33890151-a8f80b56-df52-11e7-9dae-4e46a2497e80.png">

When updating

**Comment by Kamik423** [comment]
is there a log somewhere I can post?

**Comment by nanoguo** [comment]
my solution is remove the old version manually then `brew cask install macdown`, it works for me.

**Comment by Kamik423** [comment]
By the way, got this log in `system.log` in the console:

```
Dec 29 11:58:40 HSMB15 MacDown[9151]: Sparkle: ===== MacDown =====
Dec 29 11:58:40 HSMB15 MacDown[9151]: Sparkle: Error: No valid unarchiver for /Users/<USERNAME>/Library/Caches/com.uranusjr.macdown/Sparkle/MacDown 919/YeYkOQ.html!
Dec 29 12:00:59 HSMB15 MacDown[9151]: Sparkle: Error: An error occurred while extracting the archive. Please try again later. (null) (URL (null))
```

**Comment by Kamik423** [comment]
Did some more research into that.
Found out that the versions are stored in https://macdown.uranusjr.com/sparkle/macdown/testing/appcast.xml.
There the download link for there current version is http://d.pr/f/YeYkOQ+.
When calling that link it yields the same "droplr splash screen" as the downloaded file. 
However here it vanishes soon after.
Somehow you are downloading the droplr splash screen, _not_ the linked file.
I think this is due to the hosting service.

Please fix it!

**Comment by franklinyu** [comment]
@Kamik423 Just to make sure, you intended to use the testing branch (instead of stable branch) of MacDown, right?

**Comment by Kamik423** [comment]
Yes, I used the MacDown update Menu to (try to) install a beta version, but it failed

**Comment by franklinyu** [comment]
I have no idea how the real download link looks like. The link redirects to an HTML page, and the download button provides a file located in

> https://cdn-sec.droplr.net/files/acc_49646/YeYkOQ?response-content-disposition=attachment%3B%20filename%3DMacDown.app.zip&Expires=1517342062&Key-Pair-Id=APKAJTEIOJM3LSMN33SA&Signature=fzNz3TUv-E9Tdh-ohOmKpeh-RcaYCDo95dJig2quvXG2bQQSeH0fXSD~0JlmuedrnuOvZUGwL31PuYmGpKSCH7~rUVpH2GTDnYm2NzxAo1dtrJLEabEPzD~TrW12lg8bHBaOh6SmHTcC8qbEq57iySkHUTufDGP4Cavvs0m2Hu0_

while the signature always changes. @uranusjr may have more experience on Droplr?

**Comment by Kamik423** [comment]
At least on my device I first get a „droplr splash screen“, which is the file I had downloaded, which could not be extracted. It shows while loading the page and vanishes immediately after.

**Comment by uranusjr** [comment]
There’s no way to get a consistent direct link, but the plus character at the end is supposed to make the link a direct download (instead of HTML). It seems that Droplr hates you if the file gets downloaded more than 100000 times (this is not a typo). Time to make a new dev release I guess!

**Comment by franklinyu** [comment]
Any way to get alerted by Droplr when it is downloaded 90k times? So that we can make a release before users meet this sad situation?

**Comment by franklinyu** [comment]
@Kamik423 That has been discussed in #886. In brief:

1. @uranusjr’s current testing-release workflow automatically deploys to Droplr.
2. Some users don’t have good connectivity to GitHub.

**Comment by gazcn007** [comment]
Same problem

**Comment by fuzzy76** [comment]
It's broken again.

**Comment by iptaylortechnical** [comment]
@fuzzy76 can confirm, this is not working for me.

**Comment by dreness** [comment]
This is super unfortunate, and the problem still exists (or maybe 'exists again'). I'm not sure why it's worth using any web service that's going to impose such a limitation, but maybe there are amazing benefits I don't know about that make this worthwhile. Is there a need for help re-writing the testing / release workflow, or for some other place to host the bits?

I just spent twenty minutes convincing myself this failure isn't related to any local security policies or implementations before remembering to search the interwebs. Hi, interwebs.

**Comment by rootscript** [comment]
Removing the old version (0.7.1) manually then `brew cask install macdown` does not work for me.
Instead, I downloaded from here https://macdown.uranusjr.com/history/testing/

**Comment by sgalliver** [comment]
FWIW, I can confirm the problem is happening still/again as of today.

(Thankfully, I consulted the interwebs before spending too much time debugging what I initially guessed where permissions issues.)

**Comment by FranklinYu** [comment]
@sgalliver That's #1106

**Comment by pickleshy** [comment]
Broken again! :) I love MacDown, and would like to be able to update it.
